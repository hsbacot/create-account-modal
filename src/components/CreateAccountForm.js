import React from "react";
import PropTypes from "prop-types";

export const CreateAccountForm = props => {
  const {
    email,
    password,
    handleEmail,
    handlePassword,
    handleSubmit,
    displayEmailErrors,
    formErrors
  } = props;
  const errors = formErrors.map(error => <li key={error}>{error}</li>);
  const submitEnabled = formErrors.length;
  const submitClasses = submitEnabled ? "" : "btn-default";
  return (
    <form onSubmit={handleSubmit}>
      <div className="form-section">
        <label>Email Address</label>
        <input
          name="email"
          type="text"
          value={email}
          onBlur={displayEmailErrors}
          onChange={handleEmail}
        />
      </div>
      <div className="form-section">
        <label>Password</label>
        <input
          name="password"
          type="password"
          value={password}
          onChange={handlePassword}
        />
      </div>
      {errors && <ul className="form-errors">{errors}</ul>}
      <input
        type="submit"
        value="Create Account"
        disabled={submitEnabled}
        className={`btn ${submitClasses}`}
      />
    </form>
  );
};

CreateAccountForm.propTypes = {
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  handleEmail: PropTypes.func.isRequired,
  handlePassword: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  displayEmailErrors: PropTypes.func.isRequired,
  formErrors: PropTypes.arrayOf(PropTypes.string)
};

export default CreateAccountForm;
