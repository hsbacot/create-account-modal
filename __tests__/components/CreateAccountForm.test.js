import React from "react";
import { shallow } from "enzyme";
import CreateAccountForm from "../../src/components/CreateAccountForm";

const noop = () => null;

const props = {
  email: "test@test.com",
  password: "Pa$$word!",
  handleEmail: noop,
  handlePassword: noop,
  handleSubmit: noop,
  displayEmailErrors: noop,
  formErrors: ["error one", "error two"]
};

describe("<CreateAccountForm />", () => {
  it("should render", () => {
    const component = shallow(<CreateAccountForm {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should render with no errors", () => {
    const component = shallow(<CreateAccountForm {...props} formErrors={[]} />);
    expect(component).toMatchSnapshot();
  });
});
