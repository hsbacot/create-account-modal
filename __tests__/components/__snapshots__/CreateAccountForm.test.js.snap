exports[`<CreateAccountForm /> should render 1`] = `
<form
  onSubmit={[Function]}>
  <div
    className="form-section">
    <label>
      Email Address
    </label>
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="test@test.com" />
  </div>
  <div
    className="form-section">
    <label>
      Password
    </label>
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value="Pa$$word!" />
  </div>
  <ul
    className="form-errors">
    <li>
      error one
    </li>
    <li>
      error two
    </li>
  </ul>
  <input
    className="btn "
    disabled={2}
    type="submit"
    value="Create Account" />
</form>
`;

exports[`<CreateAccountForm /> should render with no errors 1`] = `
<form
  onSubmit={[Function]}>
  <div
    className="form-section">
    <label>
      Email Address
    </label>
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="test@test.com" />
  </div>
  <div
    className="form-section">
    <label>
      Password
    </label>
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value="Pa$$word!" />
  </div>
  <ul
    className="form-errors" />
  <input
    className="btn btn-default"
    disabled={0}
    type="submit"
    value="Create Account" />
</form>
`;
